app/controllers/floating_ip_controller.rb

```diff
@@ -14,4 +14,267 @@ def self.display_methods
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Associate Floating IP \"%{name}\"") % { :name  => @floating_ip.name },
+      :url  => "/floating_ip/edit/#{@floating_ip.id}git")
+  end
+
+  def floating_ip_form_fields
+    assert_privileges("floating_ip_edit")
+    # TODO: find_by_id_filtered fails
+    # floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    floating_ip = FloatingIp.find(params[:id])
+    network_port_ems_ref = if floating_ip.network_port
+                             floating_ip.network_port.ems_ref
+                           else
+                             ""
+                           end
+    # TODO: router field is missing!
+    # :router_id            => floating_ip.router.id,
+    render :json => {
+      :fixed_ip_address     => floating_ip.fixed_ip_address,
+      :floating_ip_address  => floating_ip.address,
+      :cloud_network_name   => floating_ip.cloud_network.try(:name),
+      :network_port_ems_ref => network_port_ems_ref,
+      :cloud_tenant_name    => floating_ip.cloud_tenant.try(:name),
+    }
+  end
+
+  def get_checked_floating_ip_id(params)
+    if params[:id]
+      checked_floating_ip_id = params[:id]
+    else
+      checked_floating_ips = find_checked_items
+      checked_floating_ip_id = checked_floating_ips[0] if checked_floating_ips.length == 1
+    end
+    checked_floating_ip_id
+  end
+
+  def networks_by_ems
+    assert_privileges("floating_ip_new")
+    networks = []
+    available_networks = CloudNetwork.where(:ems_id => params[:id], :external_facing => true).find_each
+    available_networks.each do |network|
+      networks << { 'name' => network.name, 'id' => network.id }
+    end
+    render :json => {
+      :available_networks => networks
+    }
+  end
+
+  def new
+    assert_privileges("floating_ip_new")
+    @floating_ip = FloatingIp.new
+    @in_a_form = true
+    @ems_choices = {}
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
+      @ems_choices[ems.name] = ems.id
+    end
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+    drop_breadcrumb(
+      :name => _("Add New Floating IP"),
+      :url  => "/floating_ip/new"
+    )
+  end
+
+  def update
+    assert_privileges("floating_ip_edit")
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of Floating IP \"%{name}\" was cancelled by the user") % { :name  => @floating_ip.name })
+
+    when "save"
+      options = form_params
+      options.delete(:ems_id)
+      task_id = @floating_ip.update_floating_ip_queue(session[:userid], options)
+
+      add_flash(_("Floating IP update failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.to_s}, :error) unless task_id.kind_of?(Fixnum)
```

Comment:
Fixnum should be Integer now (being fixed in https://github.com/ManageIQ/manageiq/pull/13191 for the rest)